examples/src/bin/gtktest.rs

```diff
@@ -187,6 +187,8 @@ fn build_ui(application: &gtk::Application) {
 }
 
 fn main() {
+    gio::resources_register_include!("compiled.gresource").unwrap();
```

Comment:
I am a bit rusty (pun intended) in gtk, but iirc when using gtkapplication in C, registering the resources happens automatically... why can't we piggy back on the same mechanism?

Reply:
The way how that works is via shared library constructors, which are not available on all platforms (see GLib docs) and require special code.

This https://crates.io/crates/ctor crate could be used for doing that but it will probably have some impact on portability.

Reply:
Ah, I was wondering if there's a crate that supports it.

Not sure what platforms that won't work on. But I think to integrate it the Rust code using the library might still need to add a `mod` declaration to include the generated code (unless the build script compiles a separate static library that is somehow linked in...).

Reply:
Should we wait for you testing this or ... ?

Reply:
Or maybe usage of that crate could simply be documented as part of the macro?

Reply:
I'm not really sure the `ctor` crate is better here than just putting it at the start of `main()`, which shouldn't be any more code and is less magical.

Reply:
Ok!